src/generators/Minecraft.res

```diff
@@ -240,3 +240,101 @@ module MinecraftCharacter = {
     },
   }
 }
+
+type direction = [#Right | #Left | #Top | #Bottom]
+
+/* let drawCuboidLegacy = (
+  id: string,
+  source: CuboidLegacy.t,
+  dp: Builder.position,
+  ds: (int, int, int),
+  ~direction: direction=#Right,
+  (),
+) => {
+  // ~flip, ~blend, ~rotate, ~pixelate - rotate and flip can be used for some generators and for simplifying old player model drawing, respectively
+
+  /* Direction changes the placement of the back face - this replaces the ~leftSide parameter from the old branch.
+      It allows for the drawing of not only "left side" cuboids, but also any other shape that might occur.
+      In conjunction with a "~center" parameter which could specify which face of the cuboid is drawn at the center, and ~rotate and ~flip parameters that match drawTexture(),
+      any cross-shaped cuboid net should be drawable with drawCuboid().
+
+      With ~direction, and unlike ~leftSide, (x, y) will always represent the point that is (-d, -d) away from 
+      the center face's origin - the usual corner for a right direction, but not for the other directions. 
+      This makes it so that only the back face needs to be included in the switch statement.
+ */
+
+  let (x, y) = dp
+  let (w, h, d) = ds
+
+  Generator.drawTextureLegacy(id, source.front, {x: x + d, y: y + d, w: w, h: h}, ()) // Front
+  Generator.drawTextureLegacy(id, source.left, {x: x + d + w, y: y + d, w: d, h: h}, ()) // Left
+  Generator.drawTextureLegacy(id, source.right, {x: x, y: y + d, w: d, h: h}, ()) // Right
+  Generator.drawTextureLegacy(id, source.top, {x: x + d, y: y, w: w, h: d}, ()) // Top
+  Generator.drawTextureLegacy(
+    id,
+    source.bottom,
+    {x: x + d, y: y + d + h, w: w, h: d},
+    ~flip=#Vertical,
+    (),
+  ) // Bottom
+  // Back
+  switch direction {
+  | #Right =>
+    Generator.drawTextureLegacy(id, source.back, {x: x + d * 2 + w, y: y + d, w: w, h: h}, ())
+  | #Left => Generator.drawTextureLegacy(id, source.back, {x: x - d, y: y + d, w: w, h: h}, ())
+  | #Top =>
+    Generator.drawTextureLegacy(
+      id,
+      source.back,
+      {x: x + d, y: y - h, w: w, h: h},
+      ~rotateLegacy=180.0,
+      (),
+    )
+  | #Bottom =>
+    Generator.drawTextureLegacy(
+      id,
+      source.back,
+      {x: x + d, y: y + d * 2 + h, w: w, h: h},
+      ~rotateLegacy=180.0,
+      (),
+    )
+  }
+} */
+
+let drawCuboid = (
+  id: string,
+  source: Cuboid.t,
+  dp: Builder.position,
+  ds: (int, int, int),
```

Comment:
It's a good idea to give variables/arguments easy to understand names. You and I know what `dp` and `ds` mean but future coders who see this will not have the same context that we do.

Generally we should try follow [clean code principles](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29)

The principle to follow here is about [meaningful names](https://medium.com/coding-skills/clean-code-101-meaningful-names-and-functions-bf450456d90c).

Reply:
Here dp and ds are kept from the first drawCuboid() iteration - I would change them, but in my next commit I'm trying to instead change the destination to use a cuboid instead of just a position and a scale- This way parameters just for the cuboid layout can be isolated to keep drawCuboid() shorter, and separate them from the parameters carried over from drawTexture().